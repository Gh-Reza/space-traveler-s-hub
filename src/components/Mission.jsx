import PropTypes from 'prop-types';

const Mission = ({ mission }) => (
  <>
    <td>
      {mission.mission_name}
    </td>
    <td>
      {mission.description}
    </td>
    <td>
      mission status
    </td>
    <td>
      <button type="button">Join Mission</button>
    </td>
  </>
);

Mission.propTypes = {
  mission: PropTypes.string.isRequired,
};
export default Mission;
