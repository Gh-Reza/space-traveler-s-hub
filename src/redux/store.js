import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsslice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
}, applyMiddleware);

export default store;
