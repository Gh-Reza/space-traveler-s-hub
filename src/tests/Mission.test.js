import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Mission from '../components/Mission';

describe('Mission unit test', () => {
  test('Mission Renders Properly ', () => {
    const mockMission = {
      mission_id: '323333',
      mission_name: 'MockName',
      description: 'Mock detail',
      reserved: false,
    };
    const tree = renderer.create(
      <Provider store={store}>
        <Mission mission={mockMission} />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
