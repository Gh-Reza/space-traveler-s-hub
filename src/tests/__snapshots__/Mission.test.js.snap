// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission unit test Mission Renders Properly  1`] = `
Array [
  <td
    className="mission-name"
  >
    MockName
  </td>,
  <td
    className="mission-description"
  >
    Mock detail
  </td>,
  <td>
    <span
      className="status-not-active"
    >
      Not A Member
    </span>
  </td>,
  <td>
    <button
      className="join-btn"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>,
]
`;
